### PR TITLE
Micrometer: Message History factory and JMX MicrometerMessageHistory is not covered

### PR DESCRIPTION
**Pull request by jamesnetherton**
Clone of https://github.com/apache/camel-quarkus/pull/5051, rebased from latest `main` + some minor tweaks.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests/micrometer/src/test/java/org/apache/camel/quarkus/component/micrometer/it/MicrometerTest.java [hunk 1]
```diff
@@ -207,4 +207,46 @@ public void testJMXQuarkusDomain(String domain) throws Exception {
...
+
+        ObjectName on = ObjectName.getInstance(name);
+
+        //return json result
+        String json = (String) mBeanServer.invoke(on, "dumpStatisticsAsJson", null, null);
+
+        assertTrue(json.contains("jmxHistory"));
+
+    }
+
```
Comment (aldettinger) [inline integration-tests/micrometer/src/test/java/org/apache/camel/quarkus/component/micrometer/it/MicrometerTest.java:238]: Is it correct reading that `testJMXHistory` is checking that the service could be accessed through JMX ?
And that `testHistory` is ensuring that the service could be accessed through the camel context ?

If yes, what about renaming both tests ala:
 + micrometerHistoryShouldBeAvailableThroughJMX
 + micrometerHistoryShouldBeAvailableThroughCamelContext

If make sense.

Comment (jamesnetherton) [reply integration-tests/micrometer/src/test/java/org/apache/camel/quarkus/component/micrometer/it/MicrometerTest.java:238]: Yeah, if I understand the intent correctly, there's a check to ensure the relevant MBeans are published and metrics are being recorded. And another test to get it directly from the `CamelContext`. I can rename the test methods.

Maybe after #5209 these could be combined into a single test. Checking that the bits are available in JMX is probably enough for the native / non-native case.

Comment (jamesnetherton) [reply integration-tests/micrometer/src/test/java/org/apache/camel/quarkus/component/micrometer/it/MicrometerTest.java:238]: Test renamed as requested.

**Review by aldettinger — APPROVED** [review]
LGTM.

**Pull request merged**